classes/02class/exercises/c02-network07/gpavelar/README.md

```diff
@@ -0,0 +1,170 @@
+aws ec2 associate-route-table --subnet-id subnet-0f867dac87c7e75b4 --route-table-id rtb-09e3ebba72efa4af8
+# Output
+rtbassoc-08a3570a55d5db64f
+ASSOCIATIONSTATE        associated
+
+aws ec2 associate-route-table --subnet-id subnet-01ef5462d19df360c --route-table-id rtb-09e3ebba72efa4af8
+# Output
+rtbassoc-09b70548fd98b74bd
+ASSOCIATIONSTATE        associated
+
+aws ec2 associate-route-table --subnet-id subnet-01ac1852278e631c7 --route-table-id rtb-09e3ebba72efa4af8
+# Output
+rtbassoc-0019d4a663f231dee
+ASSOCIATIONSTATE        associated
+
+```
+
+- Commands for creating the following routes:
+
+| route table              | destination | target              |
+| ------------------------ | ----------- | ------------------- |
+| devopsacademy-rt-public  | 0.0.0.0     | `devopsacademy-igw` |
+| devopsacademy-rt-private | 0.0.0.0     | `devopsacademy-ngw` |
+
+```bash
+# Create route - igw
+aws ec2 create-route --route-table-id rtb-09e3ebba72efa4af8 --destination-cidr-block 0.0.0.0/0 --gateway-id igw-04518145207146fe7
+# Output
+True
+
+# Create route - ngw
+aws ec2 create-route --route-table-id rtb-040e043e6f45c836c --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0dcad8a11bc5686fa
+# Output
+An error occurred (InvalidGatewayID.NotFound) when calling the CreateRoute operation: The gateway ID 'nat-0dcad8a11bc5686fa' does not exist
+## I've already delted the natgatway created previously. But, Let's create it again
+
+### Creating NAT AGAIN
+## Allocate IP
+aws ec2 allocate-address --domain vpc
+
+# Output
+eipalloc-0711caa72a2042385      vpc     ap-southeast-2  52.62.142.50    amazon
+
+## SubnetId = subnet-08da4676baf658cb2
+## allocation-id = eipalloc-0711caa72a2042385
+## Create NAT gateway
+aws ec2 create-nat-gateway --allocation-id eipalloc-0711caa72a2042385  --subnet-id subnet-08da4676baf658cb2 --tag-specifications 'ResourceType="natgateway",Tags=[{Key="Name",Value="devopsacademy-ngw"}]'
+
+# Output
+7d2dc0a8-1b25-45ab-a711-b36b08e5f353
+NATGATEWAY      2020-07-06T15:54:11+00:00       nat-0c2e8f4ea792c8a75   pending subnet-08da4676baf658cb2        vpc-074c260483405f2bb
+NATGATEWAYADDRESSES     eipalloc-0711caa72a2042385
+TAGS    Name    devopsacademy-ngw
+
+aws ec2 delete-nat-gateway --nat-gateway-id nat-0c2e8f4ea792c8a75
+# Output
+nat-0c2e8f4ea792c8a75
+
+## Perform the command again
+aws ec2 create-route --route-table-id rtb-040e043e6f45c836c --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0c2e8f4ea792c8a75
+# Output
+True
+```
+
+- Answer the following questions:
+
+  - How did you accomplish both instructions?
+
+    ```
+    I think so.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+
+    ```
+    Through this steps I enable my subnets to be public.
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+
+    ```
+    Allow a way to connect my subnets securely over ther internet. Their IP addresses should definetely not go public. So we use NAT.
```

Comment:
Same as before, but because the machine has no public IP it needs a NAT.

The answer is the same as above, but nice if you also mention why needs to be a NGW instad of IGW.